### PR TITLE
Update versioning, introduce changelog and automate publishing

### PR DESCRIPTION
**Pull request by sgraband**
**Update theia version to latest**

Use latest version instead of next for demo applications.
Should make the apps more stable and avoids confusion,
when updating Theia Cloud next versions.

**Update components to version 0.9.0-next**

Set the version for all components to 0.9.0-next.
Reference the 0.9.0-next version for all internal dependencies.
In a follow up the builds will be adjusted so that during next builds the git-sha will be appended to the version during publishing.
Additionally, the docker images will be pushed to a version-specific next tag (e.g. 0.9.0-next).
NPM packages will be pushed with a git-sha attached and to a generic next tag, as 0.9.0-next is not a valid tag for npm packages.

Make node/landing-page a npm workspace and remove the usage of `yarn` from the READMEs and node scripts.

**Update CI to new versions and automate publishing**

Provide reuseable workflows for publishing npm packages and docker images.
Both will:
- Build on PRs for testing.
- Publish under the next tag and a specific sha version on main merges.
- Publish the version and the latest tag when a releases is published via GH.
Note that the next publishing only happens for versions that end with -next.
This way release versions (e.g. 0.9.0) are not automatically published when merged to master, instead a release is required.
This allows easy creation of patch versions with out merging them on main.

Each component has its own workflow that calls the reuseable workflows.
These workflows are only triggered when files of that component changed.

**Add version section on README + add changelog**

Introduce changelog and describe versioning in README.md.

**Helm repo PR:** https://github.com/eclipsesource/theia-cloud-helm/pull/45

Contributed on behalf of STMicroelectronics

### CODE REVIEW

File: .github/workflows/test-service.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+name: Service tests
+
+on:
```
[the remark this answers is not in the document]
Comment (sgraband) [reply .github/workflows/test-service.yml:3]: There weren't any before, but i will add the same ones as for the ci-service minus the dockerfile.

File: .github/workflows/reusable-npm.yml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+      - uses: actions/setup-node@v3
+        with:
+          node-version: "16.x"
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Build package
+        run: npm run build -w ${{ inputs.package_workspace }}
```
[the remark this answers is not in the document]
Comment (sgraband) [reply .github/workflows/reusable-npm.yml:33]: Its not the smoothest solutions but you could just append additional `-w <packageName>` to the `${{ inputs.package_workspace }}` parameter.

We could also make the input an array i believe. I will take a look if i still have time after the other improvements.

File: node/common/package.json [hunk 3]
```diff
@@ -32,7 +32,9 @@
   ],
   "scripts": {
     "build": "tsc",
-    "publish:next": "SHA=$(git rev-parse --short HEAD) && npm version prepatch --preid next.${SHA}"
+    "prepublishOnly": "npm run build",
+    "publish:next": "npm version `npm pkg get version | tr -d '\"'`.`git rev-parse --short HEAD` && npm publish --tag next",
```
[the remark this answers is not in the document]
Comment (jfaltermeier) [reply node/common/package.json:36]: The error log does not have additional information. 
I think the issue could be that `npm pkg get version` returns different results?

```
~/Git/theia-cloud/node/common$ nvm use 16
Now using node v16.20.2 (npm v8.19.4)
~/Git/theia-cloud/node/common$ npm pkg get version
{
  "@eclipse-theiacloud/common": "0.9.0-next"
}
~/Git/theia-cloud/node/common$ echo `npm pkg get version | tr -d '\"'`.`git rev-parse --short HEAD`
{ @eclipse-theiacloud/common: 0.9.0-next }.c2bca733
```

File: demo/dockerfiles/demo-theia-monitor-vscode/Dockerfile [hunk 4]
```diff
@@ -1,3 +1,3 @@
-FROM theiacloud/theia-cloud-demo:0.8.1.MS3 as production-stage
+FROM theiacloud/theia-cloud-demo:0.9.0-next as production-stage
 
-COPY --chown=theia:theia theiacloud-monitor-0.8.1.vsix /home/theia/applications/browser/plugins
\ No newline at end of file
+COPY --chown=theia:theia theiacloud-monitor-0.9.0.vsix /home/theia/applications/browser/plugins
```
[the remark this answers is not in the document]
Comment (jfaltermeier) [reply demo/dockerfiles/demo-theia-monitor-vscode/Dockerfile:3]: Either that or we try to build the vsix within this Dockerfile

File: node/common/package.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (sgraband) [reply node/common/package.json:36]: Good catch! I was using node version `v16.14.0` and the result was just `0.9.0-next`.

I just checked, from Node 18 on the output will be:

> "0.9.0-next"
> {
>   "@eclipse-theiacloud/common": "0.9.0-next"
> }

I also just checked and Node 16 is already EOL. Any objections to pin this Workflow to 16.14 and update node in a follow up?

[the remark this answers is not in the document]
Comment (jfaltermeier) [reply node/common/package.json:36]: Fine for me

File: demo/dockerfiles/demo-theia-monitor-vscode/Dockerfile (hunk 4 above)
[the remark this answers is not in the document]
Comment (sgraband) [reply demo/dockerfiles/demo-theia-monitor-vscode/Dockerfile:3]: I extended the README to be more precise on what to do for version bumps.

**Review by jfaltermeier — APPROVED** [review]

**Pull request merged**